files/create-metrics-archive

```diff
@@ -0,0 +1,17 @@
+#!/bin/sh
+
+set -e
+
+# Arguments and defaults.
+
+CURRENT_DIRECTORY=$(pwd)
+METRICS_OUTPUT_DIR=${1:-/opt/puppetlabs/puppet-metrics-collector}
+RETENTION_DAYS=${2:-30}
+OUTPUT_FILE="$CURRENT_DIRECTORY/puppet-metrics-collector-$(date +%Y.%m.%d.%H.%M.%S).tar.gz"
+
+cd "$METRICS_OUTPUT_DIR"
+find . -type f -ctime -"$RETENTION_DAYS" -and -name "*.json" -or -name "*.gz" > "$METRICS_OUTPUT_DIR.tmp"
```

Comment:
There's a couple issues with this `find`.  Because the name conditions aren't grouped, it's finding `.gz` files that are older than 30 days.

Also, because it's done inside the metrics dir it's printing the relative names beginning with `./`, so that combined with the `--transform` is producing file names like `/puppet-metrics/./puppetserver/127.0.0.1/20200213T002001Z.json`.

I'm pretty sure we can do everything in one `find | tar` command.  Let me see what I can come up with.

Reply:
Thanks. 

Note that there is a similar invocation in `files/metrics_tidy` ...

```
cd "$METRICS_DIR"
find . -type f -not -name "*.gz" -a -not -name "*.bz2" > "$METRICS_DIR.tmp"
tar -zcf "$METRICS_DIR-$(date +%Y.%m.%d.%H.%M.%S).tar.gz" --files-from "$METRICS_DIR.tmp"
xargs -a "$METRICS_DIR.tmp" rm
rm "$METRICS_DIR.tmp"
```

See also:

https://github.com/puppetlabs/puppetlabs-puppet_metrics_collector/pull/8

Reply:
I think I've addressed the finds, removed a redundant parameter from metrics-tidy, and added named parameters for create-metrics-archive!